### PR TITLE
Add deletionProtection for managed crossplane resources

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

* This PR adds deletion protection for various managed crossplane resources

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: apis/kubernetes/v1alpha1/doc.go [hunk 1]
```diff
@@ -14,9 +14,8 @@ See the License for the specific language governing permissions and
...
-
-// Hub marks this type as a conversion hub.
-func (g *Object) Hub() {
-
-}
+// Package v1alpha1 contains the v1alpha1 group Object resources of the Kubernetes provider.
+// +kubebuilder:object:generate=true
+// +groupName=kubernetes.crossplane.io
+// +versionName=v1alpha1
+package v1alpha1
```
Comment (zugao) [inline apis/kubernetes/v1alpha1/doc.go:21]: why are we moving back from v1alpha2 to v1alpha2?

Comment (Kidswiss) [reply apis/kubernetes/v1alpha1/doc.go:21]: SPKS is not yet on the latest provider-kubernetes

File: pkg/comp-functions/functions/vshnmariadb/user_management.go [hunk 2]
```diff
@@ -61,7 +61,8 @@ func addUser(comp common.Composite, svc *runtime.ServiceRuntime, username string
 				"crossplane.io/external-name": username,
 			},
 			Labels: map[string]string{
-				runtime.ProviderConfigIgnoreLabel: "true",
+				runtime.ProviderConfigIgnoreLabel:      "true",
+				"appcat.vshn.io/webhook-allowdeletion": "true",
```
Comment (Kidswiss) [inline pkg/comp-functions/functions/vshnmariadb/user_management.go:65]: Having the name of this label as a const in the `runtime` package would be nice.

**Review by Kidswiss — APPROVED** [review]
LGTM

File: apis/kubernetes/v1alpha1/doc.go (hunk 1 above)
Comment (TheBigLee) [reply apis/kubernetes/v1alpha1/doc.go:21]: we are not. we are just adding it temporarily as SPKS still uses the old provider-kubernetes

File: pkg/comp-functions/functions/vshnmariadb/user_management.go (hunk 2 above)
Comment (TheBigLee) [reply pkg/comp-functions/functions/vshnmariadb/user_management.go]: Good point. I've added a new const and replaced all labels

**Review by zugao — APPROVED** [review]
We need to test it in dev cluster before pushing to prod

**Pull request merged**